web/app/opiskeluoikeus/opiskeluoikeus.js

```diff
@@ -88,26 +100,37 @@ const filterByJärjestämislupa = (tuvaJärjestämislupa, tilat, koodiarvo) => {
   }
 }
 
-const filterBySuorituksenTyyppi = (suorituksenTyyppi, tilat, koodiarvo) => {
-  switch (suorituksenTyyppi && suorituksenTyyppi.koodiarvo) {
-    case 'vstvapaatavoitteinenkoulutus':
-      return tilat.filter((t) =>
-        vapaatavoitteisenVapaanSivistystyönKoulutuksenTilat.includes(
-          koodiarvo(t)
-        )
-      )
-    case 'vstjotpakoulutus':
-      return tilat.filter((t) =>
-        jatkuvanOppimisenVapaanSivistystyönKoulutuksenTilat.includes(
-          koodiarvo(t)
-        )
-      )
-    default:
-      return tilat.filter(
-        (t) =>
-          !vapaatavoitteisenVapaanSivistystyönKoulutuksenTilat.includes(
-            koodiarvo(t)
-          )
-      )
+const filterBySuorituksenTyyppi = (
+  opiskeluoikeudenTyyppi,
+  suorituksenTyyppi,
+  tilat,
+  koodiarvo
+) => {
+  const includedIn = (koodiarvot) => (tila) =>
+    koodiarvot.includes(koodiarvo(tila))
+  const notIncludedIn = (koodiarvot) => (tila) =>
+    !koodiarvot.includes(koodiarvo(tila))
+
+  if (opiskeluoikeudenTyyppi?.koodiarvo === 'vapaansivistystyonkoulutus') {
```

Comment:
Pohdin, että ainakin jotkut dropdown-kälikomponentit (esim. koulutusmoduulin kielivalinta jne.) osaa käyttää suoraan @KoodistoKoodiarvo-annotaation sallimia koodiarvoja. Pitäisikö tai voisiko tässä jotenkin hyväksikäyttää samaa systeemiä, niin ei tarvitsisi duplikoida sallittujen tilojen listaa käyttöliittymään?

Reply:
Yritin alkuun generisoida tämän koko homman, mutta remontti alkoi näyttämään sen verran suurelta, etten lähtenyt nyt siihen. Tässä tilapopupissa on nimittäin aivan liikaa spaghetti carbonaraa eli Bacon-spaghettia (propertyja/busseja generoidaan monella tasolla ja niiden kautta kuljetellaan tilamuutoksia eri tasoille), joten mieluiten kirjoittaisi sit koko pökäleen uusiksi. :)